test/test_integration.py

```diff
@@ -1580,3 +1581,62 @@ def hello(data):
 
         # Gets error loading dependencies.
         self.wait_for_element_by_css_selector('#output')
+
+    def test_store_type_updates(self):
+        app = dash.Dash(__name__)
+
+        types = [
+            ('str', 'hello'),
+            ('number', 1),
+            ('dict', {'data': [2, 3, 4]}),
+            ('list', [5, 6, 7]),
+            ('null', None),
+            ('bool', True),
+            ('bool', False),
+            ('empty-dict', {}),
+            ('list-dict-1', [1, 2, {'data': [55, 66, 77], 'dummy': 'dum'}]),
+            ('list-dict-1', [1, 2, {'data': [111, 99, 88]}]),
```

Comment:
Can you add the pathological items in my comments https://github.com/plotly/dash-core-components/issues/422#issuecomment-450278298 and https://github.com/plotly/dash-core-components/issues/422#issuecomment-450279988? I think you fixed all those, but here's another one to add, that I think still fails:
```js
dataChanged({a: 1, b: null}, {a: 1, c: 1}) // returns false, should be true
```

This is due to `isNil` conflating `undefined` and `null` - come to think of it, is there really a reason to treat those as equivalent? Perhaps we can just change the null check to:
```js
if (oldNull || newNull) {
    return oldData !== newData; // which will distinguish null/undefined
}
```